# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/MMPrinter.h
# Commit: acccc0eb50c9
# Full Hash: acccc0eb50c9a1111b4f152c608de7a230877385
# Author: Tom Ritter <tom@mozilla.com>
# Date: 2024-03-20 04:48:34
# Regressor Bug: 1884732
# File Overlap Count: 3
# Description:
#   Bug 1884732: Log JSActor messages with MMPrinter r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D204677
# ==============================================================================

diff -r 1957e76267e3 -r acccc0eb50c9 dom/ipc/MMPrinter.h
--- a/dom/ipc/MMPrinter.h	Tue Mar 19 17:46:39 2024 +0000
+++ b/dom/ipc/MMPrinter.h	Tue Mar 19 18:58:16 2024 +0000
@@ -21,6 +21,16 @@
     }
   }
 
+  static void Print(char const* aLocation, const nsACString& aActorName,
+                    const nsAString& aMessageName,
+                    ClonedMessageData const& aData) {
+    if (MOZ_UNLIKELY(MOZ_LOG_TEST(MMPrinter::sMMLog, LogLevel::Debug))) {
+      MMPrinter::PrintImpl(
+          aLocation,
+          NS_ConvertUTF8toUTF16(aActorName + " - "_ns) + aMessageName, aData);
+    }
+  }
+
  private:
   static LazyLogModule sMMLog;
   static void PrintImpl(char const* aLocation, const nsAString& aMsg,